src/main/conversions.c

```diff
@@ -823,9 +823,23 @@ as_status pyobject_to_val(AerospikeClient *self, as_error *err,
         Py_DECREF(py_ustr);
     }
     else if (PyBytes_Check(py_obj)) {
-        uint8_t *b = (uint8_t *)PyBytes_AsString(py_obj);
-        uint32_t b_len = (uint32_t)PyBytes_Size(py_obj);
-        *val = (as_val *)as_bytes_new_wrap(b, b_len, false);
+        char *py_obj_buffer = PyBytes_AsString(py_obj);
+        Py_ssize_t b_len = PyBytes_Size(py_obj);
+        uint8_t *new_buffer = (uint8_t *)malloc(sizeof(uint8_t) * b_len);
+        memcpy(new_buffer, py_obj_buffer, sizeof(uint8_t) * b_len);
+
+        as_bytes *bytes = as_bytes_new_wrap(new_buffer, b_len, true);
+        if (bytes == NULL) {
+            free(new_buffer);
+            return as_error_update(
+                err, AEROSPIKE_ERR_CLIENT,
+                "Unable to convert Python bytes to C client's as_bytes");
+        }
+        *val = (as_val *)bytes;
+
+        if (strstr(py_obj->ob_type->tp_name, "HyperLogLog")) {
+            bytes->type = AS_BYTES_HLL;
+        }
```

Comment:
As unlikely as it is, this might catch a user's object with HyperLogLog in the name if it inherits from bytes. I think using the full name would be better.

Reply:
Done